buildbot/master/files/config/steps.yml

```diff
@@ -22,6 +22,12 @@ mac-rel-css:
   - bash ./etc/ci/lockfile_changed.sh
   - bash ./etc/ci/manifest_changed.sh
 
+mac-nightly:
+  - ./mach build --release
+  - ./mach package --release
+  - s3cmd -c ~/.s3cfg-servo put target/release/*.tar.gz s3://servo-developer-preview/nightly/`date +%Y`/
```

Comment:
Two things:
- Prefer "$(command)" form to backticks - [style guide](https://github.com/servo/servo/blob/master/docs/STYLE_GUIDE.md).
- I believe that since make_step splits the command up into a list before passing it to Buildbot, Buildbot will exec the command directly, instead of through a shell, and the command substitution won't work. IMO it would be better to make a script for this in the servo repo in the `etc/ci` directory and call it from here.